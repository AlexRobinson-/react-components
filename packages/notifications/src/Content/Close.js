import PropTypes from 'prop-types';
import styled from 'styled-components';
import classNames from 'classnames';
import CalloutStyles from '@zendesk/garden-css-callouts/dist/index.css';
import { retrieveTheme } from 'garden-react-theming';

const Close = styled.button.attrs({
  className: props => classNames(CalloutStyles['c-callout__close'], {
    [CalloutStyles['is-focused']]: props.focused,
    [CalloutStyles['is-hovered']]: props.hovered
  })
})`
  ${props => retrieveTheme('notifications.close', props)}
`;

Close.propTypes = {
  focused: PropTypes.bool,
  hovered: PropTypes.bool
};

/** @component */
export default Close;

