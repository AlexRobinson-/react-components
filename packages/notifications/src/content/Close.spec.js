import React from 'react';
import { shallow } from 'enzyme';
import Close from './Close';

it('should render default close styling', () => {
  const wrapper = shallow(<Close />)
  expect(wrapper).toMatchSnapshot();
});

describe('state', () => {
  it('should render focused styling correctly', () => {
    const wrapper = shallow(<Close focused />)
    expect(wrapper).toMatchSnapshot();
  });

  it('should render hovered styling correctly', () => {
    const wrapper = shallow(<Close hovered />)
    expect(wrapper).toMatchSnapshot();
  });
});
