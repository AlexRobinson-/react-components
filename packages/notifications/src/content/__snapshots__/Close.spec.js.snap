// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render default close styling 1`] = `
<button
  className="Close-hpSOeC c-callout__close fyEEYD"
/>
`;

exports[`state should render focused styling correctly 1`] = `
<button
  className="Close-hpSOeC c-callout__close is-focused fyEEYD"
/>
`;

exports[`state should render hovered styling correctly 1`] = `
<button
  className="Close-hpSOeC c-callout__close is-hovered fyEEYD"
/>
`;
